src/backend/metal/src/conversions.rs

```diff
@@ -132,6 +153,661 @@ impl PrivateCapabilities {
+                    | If::STORAGE
+                    | If::COLOR_ATTACHMENT
+                    | If::COLOR_ATTACHMENT_BLEND,
+                ..defaults
+            },
+            RGBA8Snorm => Properties {
+                optimal_tiling: If::SAMPLED_LINEAR
+                    | If::STORAGE
+                    | If::COLOR_ATTACHMENT
+                    | If::COLOR_ATTACHMENT_BLEND,
+                ..defaults
+            },
+            RGBA8Uint => Properties {
+                optimal_tiling: If::STORAGE 
+                    | If::COLOR_ATTACHMENT,
+                ..defaults
+            },
+            RGBA8Sint => Properties {
+                optimal_tiling: If::STORAGE 
+                    | If::COLOR_ATTACHMENT,
+                ..defaults
+            },
+            BGRA8Unorm => Properties {
+                optimal_tiling: If::SAMPLED_LINEAR
+                    | If::STORAGE
+                    | If::COLOR_ATTACHMENT
+                    | If::COLOR_ATTACHMENT_BLEND,
+                ..defaults
+            },
+            BGRA8Unorm_sRGB if self.format_rgba8_srgb_no_write => Properties {
+                optimal_tiling: If::SAMPLED_LINEAR
+                    | If::COLOR_ATTACHMENT
+                    | If::COLOR_ATTACHMENT_BLEND,
+                ..defaults
+            },
+            BGRA8Unorm_sRGB if self.format_rgba8_srgb_all => Properties {
+                optimal_tiling: If::SAMPLED_LINEAR
+                    | If::STORAGE
+                    | If::COLOR_ATTACHMENT
+                    | If::COLOR_ATTACHMENT_BLEND,
+                ..defaults
+            },
+            RGB10A2Unorm if self.format_rgb10a2_unorm_all => Properties {
+                optimal_tiling: If::SAMPLED_LINEAR
+                    | If::STORAGE
+                    | If::COLOR_ATTACHMENT
+                    | If::COLOR_ATTACHMENT_BLEND,
+                ..defaults
+            },
+            RGB10A2Unorm if self.format_rgb10a2_unorm_no_write => Properties {
+                optimal_tiling: If::SAMPLED_LINEAR
+                    | If::COLOR_ATTACHMENT
+                    | If::COLOR_ATTACHMENT_BLEND,
+                ..defaults
+            },
+            RGB10A2Uint if self.format_rgb10a2_uint_color => Properties {
+                optimal_tiling: If::COLOR_ATTACHMENT,
+                ..defaults
+            },
+            RGB10A2Uint if self.format_rgb10a2_uint_color_write => Properties {
+                optimal_tiling: If::STORAGE 
+                    | If::COLOR_ATTACHMENT,
+                ..defaults
+            },
+            RG11B10Float if self.format_rg11b10_all => Properties {
+                optimal_tiling: If::SAMPLED_LINEAR
+                    | If::STORAGE
+                    | If::COLOR_ATTACHMENT
+                    | If::COLOR_ATTACHMENT_BLEND,
+                ..defaults
+            },
+            RG11B10Float if self.format_rg11b10_no_write => Properties {
```

Comment:
are `xx_no_write` and `xxx_all` excluding? In this case, we can omit the second `if` here, it will just drop through

Reply:
even better, since the difference is only one flag, we should move the `if` in there:
```rust
...
  | if self.format_xxx_all { If::STORAGE } else { If::empty() }
```

Reply:
Not sure I follow... If I remember right, there wasn't a pattern of feature-sets supporting certain flags, and it was rather arbitrary which is why I ended up having to have if-elseif-elseif-else in this case.